tracer/src/Datadog.Trace/ClrProfiler/Instrumentation.cs

```diff
@@ -135,16 +135,6 @@ public static void Initialize()
                 Log.Error(ex, ex.Message);
             }
 
-            try
-            {
-                var asm = typeof(Instrumentation).Assembly;
-                Log.Information($"[Assembly metadata] Location: {asm.Location}, CodeBase: {asm.CodeBase}, GAC: {asm.GlobalAssemblyCache}, HostContext: {asm.HostContext}, SecurityRuleSet: {asm.SecurityRuleSet}");
-            }
-            catch (Exception ex)
-            {
-                Log.Error(ex, ex.Message);
-            }
-
```

Comment:
This doesn't seem related to the framework upgrade? 🤔

Reply:
The reason was `asm.CodeBase` and `asm.GlobalAssemblyCache` are obsolete in net5+. I think this log could still be useful, so I `ifdef`'d the message